File: Java/JCrimsonEdge/CeCore/src/main/java/org/crimsonedge/core/entity/JavaState.java
package org.crimsonedge.core.entity;

import org.crimsonedge.core.CeClass;

@CeClass("cryo::csJavaState")
public class csJavaState extends csEntityState{

    public csJavaState() {
    }

    public csJavaState(long ref) {
        super(ref);
    }

    public void onAttachedToWorld(csWorld world) {
    }

    public void onDetachedFromWorld(csWorld world) {
    }

    public void update(float tpf) {

    }

    public csSpatialState getRoot () {
        csEntity entity = getEntity();
        return entity != null ? entity.getRoot() : null;
    }
}

File: Engine/ceCore/entity/javastate.cc

#include <ceCore/entity/javastate.hh>
#include <ceCore/java.hh>
#include <ceCore/java_methods.hh>
#include <ceCore/entity/world.hh>

#define THIS_CLASS_NAME  "org/crimsonedge/core/entity/JavaState"

namespace cryo
{


JavaState::JavaState()
: EntityState()
{
  SetNeedUpdate(true);
}

void JavaState::OnAttachedToWorld(cryo::World *world)
{
  static JavaCallVoid1<jobject> onAttachedToWorld (Java::Get(), this, THIS_CLASS_NAME, "onAttachedToWorld", "Lorg/crimsonedge/core/entity/World;");
  onAttachedToWorld.call(Java::Get(), world->GetJObject());
}


void JavaState::OnDetachedFromWorld(cryo::World *world)
{
  static JavaCallVoid1<jobject> onDetachedFromWorld (Java::Get(), this, THIS_CLASS_NAME, "onDetachedFromWorld", "Lorg/crimsonedge/core/entity/World;");
  onDetachedFromWorld.call(Java::Get(), world->GetJObject());
}


void JavaState::Update(float tpf)
{
  static JavaCallVoid1<jfloat> update (Java::Get(), this, THIS_CLASS_NAME, "update", JAVA_FLOAT);
  update.call(Java::Get(), tpf);
}

}
